src/main/java/edu/kit/kastel/sdq/artemis4j/grading/Assessment.java

```diff
@@ -90,7 +90,7 @@ public Assessment(
                 result.feedbacks());
 
         // Unpack the result
-        this.annotations = MetaFeedbackMapper.parseMetaFeedbacks(feedbacks, config);
```

Comment:
Artemis4J is not thread safe. (As far as I know.) Thus, the handling of synchronization should not be in artemis4J :)

Reply:
Yes, Artemis4j isn't totally thread-safe. However, some limited thread-safety for adding and removing annotations makes IntelliGrade *significantly* simpler (there, the Autograder needs to be executed in a background thread to not block the UI)

Reply:
Ok .. I just wanted to ensure that artemis4j does not need to handle all synchronization.